app/controllers/articles_controller.rb

```diff
@@ -1,17 +1,18 @@
 class ArticlesController < ApplicationController
-
- before_action :set_article, only: [:show, :edit, :update, :destroy]
- http_basic_authenticate_with name: "dhh", password: "secret", except: [:index, :show]
-
+  before_action :authenticate_user!
+  before_action :set_article, only: [:show, :edit, :update, :destroy]
+  http_basic_authenticate_with name: "dhh", password: "secret", except: [:index, :show]
```

Comment:
I think user is now supposed to login with password. So, we don’t need this.

Reply:
Updated